### PR TITLE
Add options for retina logo and display site title and tagline

### PR DESCRIPTION
**Pull request by carolinan**
Add a customizer option for a retina logo
Add a customizer option to display or hide the site title and tagline.

replaces customizer.php with a new file, classes\class-twenty-twenty-one-customize.php

### CODE REVIEW

**Comment by melchoyce** [comment]
I'm not sure the retina setting is working. I'm not seeing a quality change when I select it, the logo remains kind of fuzzy: https://cloudup.com/ce6z4EbZbHn

**Comment by melchoyce** [comment]
Actually it doesn't look like `Display Site Title & Tagline` is working, either; selecting the checkbox doesn't change anything for me.

**Comment by carolinan** [comment]
> Actually it doesn't look like `Display Site Title & Tagline` is working, either; selecting the checkbox doesn't change anything for me.

Yep, it is missing two important changes, I forgot to copy over all the files to the github client. Will fix when I get to the computer next.

File: template-parts/header/site-branding.php [hunk 1]
```diff
@@ -15,8 +20,8 @@
 		<?php
 	}
 	$blog_info = get_bloginfo( 'name' );
-	if ( ! empty( $blog_info ) ) {
-		if ( is_front_page() || ( is_front_page() && is_home() ) ) {
+	if ( ! empty( $blog_info ) && get_theme_mod( 'display_title_and_tagline', true ) === true ) {
```
Comment (carolinan) [inline template-parts/header/site-branding.php:23]: this can be simplified

**Comment by carolinan** [comment]
I do not know how to fix the PHPStan errors.

File: classes/class-twenty-twenty-one-customize.php [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	 * CUSTOMIZER SETTINGS
+	 */
+	class Twenty_Twenty_One_Customize {
+
+		/**
+		 * Register customizer options.
+		 *
+		 * @param WP_Customize_Manager $wp_customize Theme Customizer object.
+		 */
+		public static function register( $wp_customize ) {
```
Comment (pattonwebz) [inline classes/class-twenty-twenty-one-customize.php]:
```suggestion
		/**
		 * Register customizer options.
		 *
		 * @param WP_Customize_Manager $wp_customize Theme Customizer object.
		 *
		 * @return void
		 */
		public static function register( $wp_customize ) {
```

File: classes/class-twenty-twenty-one-customize.php [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+		}
+
+		/**
+		 * Sanitize boolean for checkbox.
+		 *
+		 * @param bool $checked Whether or not a box is checked.
+		 * @return bool
+		 */
+		public static function sanitize_checkbox( $checked ) {
```
Comment (pattonwebz) [inline classes/class-twenty-twenty-one-customize.php]:
```suggestion
		public static function sanitize_checkbox( $checked = null ) {
```

**Comment by pattonwebz** [comment]
@carolinan I added 2 suggestions that I think fix the phpstan errors :)

File: classes/class-twenty-twenty-one-customize.php [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+				array(
+					'type'    => 'checkbox',
+					'section' => 'title_tagline',
+					'label'   => __( 'Display Site Title & Tagline', 'twentytwentyone' ),
+				)
+			);
+
+			// Background color.
+			// Include the custom control class.
+			include_once get_theme_file_path( 'classes/class-twenty-twenty-one-customize-color-control.php' );
```
Comment (pattonwebz) [inline classes/class-twenty-twenty-one-customize.php]:
```suggestion
			include_once get_theme_file_path( 'classes/class-twenty-twenty-one-customize-color-control.php' ); // phpcs:ignore WPThemeReview.CoreFunctionality.FileInclude.FileIncludeFound
```

**Review by aristath — APPROVED** [review]
Code looks good and everything seems to work.
I think this is OK to merge. If there are retina issues (I was unable to test that), a separate issue can be opened to address it :+1:

**Pull request merged**